src/vs/base/parts/tree/browser/treeModel.ts

```diff
@@ -980,19 +980,28 @@ export class TreeModel {
 	}
 
 	public refresh(element: any = null, recursive: boolean = true): WinJS.Promise {
-		var item = this.getItem(element);
+		var item: Item = this.getItem(element);
 
 		if (!item) {
 			return WinJS.TPromise.as(null);
 		}
 
+		this.applyChange(item.getElement(), element);
+
 		var eventData: IRefreshEvent = { item: item, recursive: recursive };
 		this._onRefresh.fire(eventData);
 		return item.refresh(recursive).then(() => {
 			this._onDidRefresh.fire(eventData);
 		});
 	}
 
+	private applyChange(target: any, source: any): void {
```

Comment:
if it's a bug in vs, ca  you report it to them or even fix it there?

Reply:
I already sent emails for reporting this bug to you and @kburtram . Let me create a bug for this now.